### PR TITLE
Change the signature for DispatchData.enumerateBytes() to match that …

### PR DESCRIPTION
**Pull request by ktopley-apple**
…of Data.enumerateBytes() for Swift 4.2 and obsolete the old version.

rdar://problem/40775762

Adds overload DispatchData.enumerateBytes(_:) to match the signature used in Data and make it available as of Swift 4.2. Also obsoletes the existing version.

Resolves [SR-7868](https://bugs.swift.org/browse/SR-7868).

### CODE REVIEW

**Comment by ktopley-apple** [comment]
@swift-ci: please smoke test

**Comment by jrose-apple** [comment]
The code looks good. I wonder if it's worth adding an explicit test that runs in 4.2 mode, so that you can check that the new signature is working too.

**Comment by ktopley-apple** [comment]
@swift-ci: please test

File: test/stdlib/DispatchData.swift [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+		nextIndex = offset + ptr.count
+	})
+#else
+	data.enumerateBytes(block: { ptr, offset, stop in
+		for i in 0..<ptr.count {
+			expectEqual(ptr[i + offset], UInt8(i + offset))
+		}
+		nextIndex = offset + ptr.count
+	})
+#endif
```
Comment (jrose-apple) [inline test/stdlib/DispatchData.swift]: I suggest a trailing closure test that's expected to work in both language modes, too.

**Review by jrose-apple — APPROVED** [review]

File: test/stdlib/DispatchData.swift (hunk 1 above)
Comment (ktopley-apple) [reply test/stdlib/DispatchData.swift]: Added that test and squashed commits.

**Comment by ktopley-apple** [comment]
@swift-ci please test

**Comment by ktopley-apple** [comment]
@swift-ci please smoke test

**Comment by jrose-apple** [comment]
Note for next time: the regular tests include the smoke tests, so you don't actually have to run both.

**Review by mwwa — APPROVED** [review]

**Pull request merged**